scripts/lib/robot.sh

```diff
@@ -139,6 +139,29 @@ execute_robot() {
     installed_dut_option=""
   fi
 
+  # Prevent executing tests on a dirty git tree
+
+  if [[ -z "${ALLOW_DIRTY}" ]]; then
+    echo "Checking if running the test from a reproducible revision."
+    echo "If NECESSARY, set ALLOW_DIRTY to skip the check."
+
+    if ! git diff --quiet || ! git diff --staged --quiet; then
```

Comment:
Just a note that it does not include untracked files. Not sure if intended or not. Typically, we call a repo -dirty if there are new untrackd files as well.

Reply:
This doesn't seem to catch changes in submodules.

Reply:
Never mind, it just doesn't catch stuff like `modified:   3rdparty/dasharo-blobs (untracked content)`, which is the same as above, but in submodule. `(modified content)` is caught properly.